src/zap.c

```diff
@@ -5002,7 +5002,8 @@ destroy_one_item(struct obj *obj, int osym, int dmgtyp)
         quan = obj->quan;
         switch (osym) {
         case RING_CLASS:
-            if (obj->otyp == RIN_SHOCK_RESISTANCE) {
+            if (obj->otyp == RIN_SHOCK_RESISTANCE ||
+                    (obj->owornmask && uarmg && !is_metallic(uarmg))) {
```

Comment:
`obj->owornmask` will also have a value for stuff like wielded objects, so this would protect rings you have quivered, wielded, or readied in your alternate weapon slot.  `(obj->owornmask & W_RING)` would check more specifically if the item is being worn in one of the ring slots.